source/Handlebars/HandlebarsEnvironment.cs

```diff
@@ -12,7 +12,24 @@ public HandlebarsEnvironment(HandlebarsConfiguration configuration)
             Configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
         }
 
-        public Func<object, string> CompileView(string templatePath)
+        public Action<TextWriter, object> CompileWriterView(string templatePath)
+        {
+            return CompileViewInternal(templatePath);
+        }
+        public Func<object,string> CompileView(string templatePath)
+        {
+            var view = CompileViewInternal(templatePath);
+            return (vm) =>
+            {
+                using (var writer = new PolledStringWriter(Configuration.FormatProvider))
+                {
+                    view(writer, vm);
+                    return writer.ToString();
+                }
+            };
+        }
+
+        private Action<TextWriter, object> CompileViewInternal(string templatePath)
```

Comment:
In order to stay consistent with the current interface I'd suggest doing something like this:
```csharp
public Action<TextWriter, object> CompileView(string templatePath, Func<string, TextReader> readerFactory)
{
    return CompileViewInternal(templatePath, readerFactory);
}

public Func<object,string> CompileView(string templatePath)
{
    TextReader ReaderFactory(string template) => new StringReader(template); 
    
    var view = CompileViewInternal(templatePath, ReaderFactory);
    return (vm) =>
    {
        using (var writer = new PolledStringWriter(Configuration.FormatProvider))
        {
            view(writer, vm);
            return writer.ToString();
        }
    };
}

private Action<TextWriter, object> CompileViewInternal(string templatePath, Func<string, TextReader> readerFactory)
{
    var configuration = new InternalHandlebarsConfiguration(Configuration);
    var createdFeatures = configuration.Features;
    for (var index = 0; index < createdFeatures.Count; index++)
    {
        createdFeatures[index].OnCompiling(configuration);
    }

    var compiler = new HandlebarsCompiler(configuration);
    var compiledView = compiler.CompileView(readerFactory, templatePath, configuration);
    
    for (var index = 0; index < createdFeatures.Count; index++)
    {
        createdFeatures[index].CompilationCompleted();
    }

    return compiledView;
}
``` 
You'd still need to fix `HandlebarsCompiler.CompileView(...)` method by passing `Func<string, TextReader> readerFactory` and using it instead of `new StringReader(template)`.